channeldb/db.go

```diff
@@ -511,7 +511,7 @@ func fetchChannels(d *DB, pending, waitingClose bool) ([]*OpenChannel, error) {
 						"node_key=%x: %v", chainHash[:], k, err)
 				}
 				for _, channel := range nodeChans {
-					if channel.IsPending != pending {
+					if !waitingClose && channel.IsPending != pending {
```

Comment:
Hmm.. this code is not straight forward to follow. Would it be better to instead add the channel if either
1) `channel.IsPending == pending`
2) `channelWaitingClose == waitingClose`

Another option is to only consider the `pending` argument if it is true, and use both `pending` and `waitingClose` as a way of filter channels.

Reply:
I'm not sure either is equivalent to the new behavior? Switching from 1 && 2 to 1 || 2 doesn't offer enough constraint to target a single quadrant of the state space. The new behavior ignores `pending` if we are looking for `waitingClose=true` channels, which is not the same as only considering `pending` if `pending=true`.

Docs + comments would help out here for sure

Reply:
yeah, but with this ther will also be a quadrant impossible to target? (`1 && 2`)

Reply:
The simplest change (the one proposed) is to treat channels that are pending and waiting close as waiting close, so it’s collapsing the two quadrants where waiting close is true. 

We could leave the filtering logic as is, and just query for (pending=true, waitingClose=true) and (pending=false, waitingClose=true) and combine the results, which would be equivalent

Reply:
Latter suggestion SGTM :)

Reply:
Fixed.